client/galaxy/scripts/mvc/tool/tool-form-base.js

```diff
@@ -38,9 +53,7 @@ define(['utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view',
                 operations      : !this.options.hide_operations && this._operations(),
                 onchange        : function() {
                     self.deferred.reset();
```

Comment:
The reset here should probably be placed into the refresh call. It removes uneccessarily queued requests.